Suggested changes for ConcurrentTimedRotatingFileHandler.

Preston-Landers opened the pull request:
Here are some suggested changes for this branch.

Cleanup for TimedRotatingFileHandler.
The shouldRollover method is defined in the TimedRotatingFileHandler base class.
Also fix up the ConcurrentTimedRotatingFileHandler `__init__` so that all arguments are handled.

Keeping BaseRotatingHandler as the base class for ConcurrentFileHandlerMixin helps avoid a bunch of unknown property warnings.

WARNING: I have NOT done extensive testing of ConcurrentTimedRotatingFileHandler!

Commit:
Implement ConcurrentTimedRotatingFileHandler

Move the current implementation of RotatingFileHandler to a mixin (uside
the shouldRollover method).

Use the mixin to patch the new implementations into the two subclasses
(ConcurrentTimedRotatingFileHandler and ConcurrentRotatingFileHandler).

Move the specific shouldRollover logic to the
ConcurrentRotatingFileHandler implementation.

aDramaQueen committed:
see CHANGELOG.md v.0.9.21

Preston-Landers committed:
Cleanup for TimedRotatingFileHandler.
The shouldRollover method is defined in the TimedRotatingFileHandler base class.
Also fix up the ConcurrentTimedRotatingFileHandler `__init__` so that all arguments are handled.

Preston-Landers commented:
Intended as an update for:
https://github.com/Preston-Landers/concurrent-log-handler/pull/48

tshirtman commented on src/concurrent_log_handler/__init__.py:
```diff
@@ -573,10 +575,28 @@ def _shouldRollover(self):
+        owner=None, chmod=None, umask=None, newline=None, terminator="\n",
+        unicode_error_policy='ignore'
+    ):
+        TimedRotatingFileHandler.__init__(
```
you shouldn’t need to call init methods explicitely in a specific order, python’s `mro` ensures that `super()` will resolve these in order, and that multiple inheritance works. https://rhettinger.wordpress.com/2011/05/26/super-considered-super/

Preston-Landers replied:
I think the problem there is the different method signatures (arguments) on TimedRotatingFH vs ConcurrentFHMixin. Your link points out the need for the method signatures to align. While there might be ways to use `**kwargs` creatively here, I tend to prefer explicitly naming all the arguments in each place. Method resolution order wasn't the problem, it's the incompatibility of the method signatures.

tzongw committed:
fix doRollover(missing 1 file to rename)

Preston-Landers committed:
Merge pull request #52 from tzongw/master

fix doRollover(missing 1 file to rename)

Preston-Landers committed:
Option for stresstest.py to use ConcurrentTimedRotatingFileHandler.
Use the --timed=10 option to use a 10 second timed rotation.
Package __all__ should include ConcurrentTimedRotatingFileHandler.

Preston-Landers committed:
Make stresstest.py automatically delete the test files at the start of the run
Unless the -k / --keep option is given, in which case the test will fail if the files already exist.

Preston-Landers committed:
Updates to the new `lock_file_directory` setting.
Add some comments about it to the README.md.
Bump the version to 0.9.21.
Add support for `lock_file_directory` to stresstest.py.

Preston-Landers committed:
Add `lock_file_directory` kwarg to ConcurrentTimedRotatingFileHandler

Preston-Landers committed:
Cleanup Changelog markdown.
Add note about PR 52.

Commit:
add namer support

Preston-Landers committed:
Merge pull request #54 from dashedman/namer_support

add support for custom namer function

Preston-Landers committed:
Mention the `namer` attribute in the README.md.

Preston-Landers committed:
Add a little more info about concurrency to the README.

Preston-Landers committed:
Enhanced test suite.

GPT-4 provided inspiration in several areas.

Preston-Landers committed:
Apply `black` formatting.

Preston-Landers committed:
Readme fix and GH Workflows fix.

Preston-Landers committed:
GH Workflows fixes.

Preston-Landers committed:
GH Workflows fixes.

Preston-Landers committed:
GH Workflows fixes.

Preston-Landers committed:
Suppress ruff errors on Python 2 lines.

Preston-Landers committed:
Suppress ruff errors on Python 2 lines.

Preston-Landers committed:
Suppress ruff errors on long lines.

Preston-Landers committed:
Add ruff config file.

Preston-Landers committed:
Fix README notes about uploading new packages.

Preston-Landers committed:
Fix f-string usage breaking Python 2.7 support.

However, this (0.9.22) will be the LAST version to support Python 2!

Preston-Landers committed:
Remove relics of Python 2 support.
Update version to 0.9.23.
Require Python 3.6 or higher, mainly due to f-string support.
